src/components/page-step/index.tsx

```diff
@@ -0,0 +1,279 @@
+import { getStep } from '../../state/job';
+import { getJobStepLog } from '../../state/job-logs';
+
+const isStepRunning = (step: any) => step && !step.ended && step.started;
+
+export interface PagePipelineStepsSubscription {
+  subscribe: (props: string, string) => void;
+  unsubscribe: (props: string, string) => void;
+}
+
+export interface PageStepsProps extends PagePipelineStepsSubscription {
+  appName: string;
+  jobName: string;
+  step: StepModel;
+  pipelineRuns: Array<PipelineRunModel>;
+  stepName: string;
+  stepLog?: string;
+}
+
+export class PageStep extends Component<PageStepsProps, { now: Date }> {
+  static readonly propTypes: PropTypes.ValidationMap<PageStepsProps> = {
+    appName: PropTypes.string.isRequired,
+    jobName: PropTypes.string.isRequired,
+    step: PropTypes.shape(
+      StepModelValidationMap
+    ) as PropTypes.Requireable<StepModel>,
+    stepName: PropTypes.string.isRequired,
+    stepLog: PropTypes.string,
+    pipelineRuns: PropTypes.arrayOf(
+      PropTypes.shape(
+        PipelineRunModelValidationMap
+      ) as PropTypes.Validator<PipelineRunModel>
+    ),
+    subscribe: PropTypes.func.isRequired,
+    unsubscribe: PropTypes.func.isRequired,
+  };
+  private interval;
+
+  constructor(props: PageStepsProps) {
+    super(props);
+    this.state = { now: new Date() };
+  }
+
+  override componentDidMount() {
+    const { subscribe, appName, jobName } = this.props;
+    subscribe(appName, jobName);
+    this.interval = setInterval(() => this.setState({ now: new Date() }), 1000);
+  }
+
+  override componentWillUnmount() {
+    const { unsubscribe, appName, jobName } = this.props;
+    unsubscribe(appName, jobName);
+    clearInterval(this.interval);
+  }
+
+  override componentDidUpdate(prevProps: PageStepsProps) {
+    const { subscribe, unsubscribe, appName, jobName, step } = this.props;
+
+    if (prevProps.jobName !== jobName || prevProps.appName !== appName) {
+      unsubscribe(appName, prevProps.jobName);
+      subscribe(appName, jobName);
+    }
+
+    this.configureTimerInterval(step);
+  }
+
+  configureTimerInterval(step) {
+    clearInterval(this.interval);
+    if (isStepRunning(step)) {
+      this.interval = setInterval(
+        () => this.setState({ now: new Date() }),
+        1000
+      );
+    }
+  }
+
+  override render() {
+    const { appName, jobName, stepName, step } = this.props;
+    return (
+      <>
+        <DocumentTitle title={stepName} />
+        <Breadcrumb
+          links={[
+            { label: appName, to: routeWithParams(routes.app, { appName }) },
+            {
+              label: 'Pipeline Jobs',
+              to: routeWithParams(routes.appJobs, { appName }),
+            },
+            {
+              label: smallJobName(jobName),
+              to: routeWithParams(routes.appJob, { appName, jobName }),
+            },
+            { label: getPipelineStepDescription(stepName) },
+          ]}
+        />
+        {!this.props.step ? (
```

Comment:
use step variable declared above instead
```suggestion
        {!step ? (
```